engine/runtime/src/main/scala/org/enso/compiler/data/BindingsMap.scala

```diff
@@ -191,12 +191,31 @@ case class BindingsMap(
     )
   }
 
+  def resolveQualifiedNameIn(
```

Comment:
I think it would be really helpful if this method had a doc comment stating what is its purpose. By just looking at it, I don't understand what is meant to happen here - we have some non-trivial control flow (`return` from a `for`, something rather very odd for Scala-style) and I'm not sure if it was intended because I don't know what the function is supposed to do. Where the `submoduleNames` come from?

We seem to be checking all submodule names for errors but keeping only the last one for the result. Why? Why last one not the first one for example? It's all unclear.

Reply:
I see that this logic was here earlier, so I understand it's not directly part of your change. Still if it is possible to add some comment explaining it, I think it would really help us in the future.

Reply:
Okay doing some further reading I think I now know why we are using the last one. Still not sure why error-processing is done the way it is (not saying it's wrong, I just don't know).
Still - if we can clarify it, it would be extremely helpful in checking what this helper does and if it indeed does the right thing.